### PR TITLE
Add RegionOperationProcedure

### PR DESCRIPTION
**Pull request by liyuheng55555**

### CODE REVIEW

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/RegionOperationProcedure.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class RegionOperationProcedure<TState>
+    extends StateMachineProcedure<ConfigNodeProcedureEnv, TState> {
+
+  // regionId -> regionOperationProcedure id
+  private static final Map<TConsensusGroupId, Integer> regionOperationsMap = new HashMap<>();
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/RegionOperationProcedure.java:37]: concurrenthashmap？

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/RegionOperationProcedure.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  private static synchronized void addToMap(TConsensusGroupId consensusGroupId, long procId) {
+    regionOperationsMap.putIfAbsent(consensusGroupId, 0);
+    regionOperationsMap.compute(consensusGroupId, (k, v) -> v + 1);
+  }
+
+  @Override
+  protected void procedureEndHook() {
+    synchronized (RegionOperationProcedure.class) {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/RegionOperationProcedure.java:58]: do not need this if we use concurrenthashmap?

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/ProcedureManager.java [hunk 3]
```diff
@@ -637,14 +625,32 @@ private TSStatus checkRegionMigrate(
         && ConsensusFactory.SIMPLE_CONSENSUS.equals(conf.getSchemaRegionConsensusProtocolClass())) {
       failMessage =
           "The region you are trying to migrate is using SimpleConsensus, and SimpleConsensus not supports region migration.";
-    } else if (anotherMigrateProcedure.isPresent()) {
+    } else if (0 != RegionOperationProcedure.getRegionOperations(regionGroupId)) {
+      Optional<Procedure<ConfigNodeProcedureEnv>> anotherMigrateProcedure =
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/ProcedureManager.java:629]: why need to  traversal? Cannot we use `RegionOperationProcedure.getRegionOperations(regionGroupId)` to get is directly?

**Comment by wangchao316** [comment]
hi, why add this procedure ?  function is ?

**Comment by liyuheng55555** [comment]
> hi, why add this procedure ? function is ?
@wangchao316 
Hi, the main purpose is to maintenance the RegionOperationProcedure.regionOperationsMap. And this map is for temporarily stopping the auto leader balancer during region operation. But for now we might find a better way, this pr may be closed later.

**Comment by OneSizeFitsQuorum** [comment]
Will not do this for maintainability, close it now

**Pull request closed without merging**